Python/thread_pthread.h

```diff
@@ -462,16 +462,27 @@ PyThread_acquire_lock_timed(PyThread_type_lock lock, PY_TIMEOUT_T microseconds,
     }
 
     _PyTime_t deadline = 0;
-    if (timeout > 0 && !intr_flag) {
+    if (timeout > 0
+#ifndef HAVE_SEM_CLOCKWAIT
+        && !intr_flag
+#endif
+        )
+    {
         deadline = _PyTime_GetMonotonicClock() + timeout;
     }
 
     while (1) {
         if (timeout > 0) {
+#ifdef HAVE_SEM_CLOCKWAIT
+            struct timespec abs_timeout;
+            _PyTime_AsTimespec_clamp(deadline, &abs_timeout);
```

Comment:
I don't know if sem_clockwait() timeout argument can be modified by the function, for example if the wait is interrupted by a signal. In case of doubt, I prefer to compute abs_timeout before each call.

Reply:
In the same file, pthread_cond_timedwait() absolute timeout is only computed once **outside** the loop. I understand that we can also compute abs_timeout once outside the loop.

The new implementation of time.sleep(), pysleep() function, now also uses an absolute timeout (timespec) computed once outside the loop. It uses clock_nanosleep() or nanosleep(). These functions have an argument for the remaining time, it's not used by pysleep().

Reply:
sem_clockwait() documentation doesn't say that the absolute timeout is modified, which confirms that it can be only computed once:
https://www.gnu.org/software/libc/manual/html_node/Waiting-with-Explicit-Clocks.html